src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int POSITION_FOR_NULL_KEYS = 0;
+    private int size;
+    private int mapCapacity;
+    private Node<K, V>[] table;
+    private int threshold;
+
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            table = resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (key == null) {
```

Comment:
why do we need special logic for `key == null`? Could we just treat with this case same way as for `key != null`